WalkingChallenge/Settings/TeamSettingsDataSource.swift

```diff
@@ -32,17 +32,77 @@ import Foundation
 class TeamSettingsDataSource: TableViewDataSource {
   var cells: [[CellContext]] = []
 
+  private var teamName: String = " "
+  private var eventName: String = " "
+  private var team: (members: [(name: String?, image: URL?)], capacity: Int) = ([], 0)
+
+  func reload(completion: @escaping () -> Void) {
+    configure()
+    completion()
+
+    onBackground { [weak self] in
+      let CC: NSCondition = NSCondition()
+      var MS: DispatchSemaphore = DispatchSemaphore(value: 0)
+
+      var capacity: Int = 1
+      var members: [(name: String?, image: URL?)] = []
+
+      AKFCausesService.getParticipant(fbid: Facebook.id) { (result) in
+        guard let participant = Participant(json: result.response) else { return }
+
+        if let event = participant.event?.id {
+          AKFCausesService.getEvent(event: event) { (result) in
+            guard let event = Event(json: result.response) else { return }
+            capacity = event.teamLimit
+            CC.signal()
+          }
+        }
+
+        if let team = participant.team {
+          self?.teamName = team.name ?? " "
+
+          if let team = participant.team?.members {
+            MS = DispatchSemaphore(value: team.count * 2)
+            members = Array<(String?, URL?)>.init(repeating: (name: nil, image: nil), count: team.count)
+            for (index, member) in team.enumerated() {
+              Facebook.getRealName(for: member.fbid) { (name) in
+                members[index].name = name
+                MS.signal()
```

Comment:
the only issue I can see with this is if Facebook fails the signal won't trigger

Reply:
Ugh, I left the movement on my local tree.